hackathon_site/event/api_views.py

```diff
@@ -66,3 +67,41 @@ def post(self, request, *args, **kwargs):
         response_serializer = TeamSerializer(profile.team)
         response_data = response_serializer.data
         return Response(data=response_data, status=status.HTTP_201_CREATED,)
+
+
+class JoinTeamView(generics.GenericAPIView, mixins.RetrieveModelMixin):
+    permission_classes = [UserHasProfile]
+    serializer_class = TeamSerializer
+    lookup_field = "team_code"
+    queryset = Team
+
+    @transaction.atomic
+    def post(self, request, team_code, *args, **kwargs):
+
+        profile = request.user.profile
+        current_team = profile.team
+
+        self.kwargs["lookup_url_kwarg"] = team_code
+        team = self.get_object()
+
+        if team.profiles.count() >= Team.MAX_MEMBERS:
+            raise ValidationError({"detail": "Team is full."})
```

Comment:
Minor for consistency:

```suggestion
            raise ValidationError({"detail": "Team is full"})
```